code/game/objects/items/weapons/twohanded.dm

```diff
@@ -303,8 +303,67 @@
 	hitsound = 'sound/weapons/bladeslice.ogg'
 	attack_verb = list("attacked", "poked", "jabbed", "torn", "gored")
 	sharpness = IS_SHARP
+	var/obj/item/weapon/grenade/explosive = null
+	var/war_cry = "AAAAARGH!!!"
 
 /obj/item/weapon/twohanded/spear/update_icon()
-	icon_state = "spearglass[wielded]"
-	return
+	if(explosive)
+		icon_state = "spearbomb[wielded]"
+	else
+		icon_state = "spearglass[wielded]"
+
+/obj/item/weapon/twohanded/spear/afterattack(atom/movable/AM, mob/user, proximity)
+	if(!proximity)
+		return
+	if(istype(AM, /turf/simulated/floor)) //So you can actually melee with it
+		return

```

Comment:
Just remove the second argument of istype so it will check the AM to be atom/movable
Or exploding when hitting space is intended?